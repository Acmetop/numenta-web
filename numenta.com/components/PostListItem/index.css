.columns {
  composes: layout-columns from '../../static/assets/css/theme.css';
}

.content {
  composes: layout-content from '../../static/assets/css/theme.css';
}

.aside {
  composes: mt1 from 'tachyons';
  composes: tc from 'tachyons';
  composes: layout-aside from '../../static/assets/css/theme.css';
}


.postListItem {
  composes: b--light-gray from 'tachyons';
  composes: bt from 'tachyons';
  composes: pb1 from 'tachyons';
  composes: pt3 from 'tachyons';
}

.date {
  composes: fw5 from 'tachyons';
  composes: lh-copy from 'tachyons';
  composes: mb2 from 'tachyons';
}

.title {
  composes: db from 'tachyons';
  composes: f4 from 'tachyons';
  composes: lh-title from 'tachyons';
  composes: mb1 from 'tachyons';
}

.author {
  composes: lh-copy from 'tachyons';
  composes: relative from 'tachyons';
  top: -0.5rem;
}

.image {
  composes: center from 'tachyons';
  composes: w-80 from 'tachyons';
  composes: w-auto-m from 'tachyons';
}
