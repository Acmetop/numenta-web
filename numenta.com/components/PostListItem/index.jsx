// Numenta company website source code. Copyright Â© 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import moment from 'moment'
import React from 'react'

import {getEventTimeDisplay} from '../../utils/shared'

import Avatar from '../Avatar'
import Paragraph from '../Paragraph'
import Spacer from '../Spacer'
import Strong from '../Strong'
import Subtle from '../Subtle'
import Table from '../Table'
import TableBody from '../TableBody'
import TableCell from '../TableCell'
import TableRow from '../TableRow'
import TextLink from '../TextLink'
import Time from '../Time'

import styles from './index.css'


/**
 *
 */
const PostListItem = ({post}, {config}) => {
  const {data, file, path} = post
  const datetime = moment(data.date, config.moments.post)
  const occur = datetime.format(config.moments.human)
  const target = (data.type === 'link') ? data.link : path
  const key = file.dir.split('/')[0]
  let brief = (<Paragraph>{data.brief}...</Paragraph>)

  // modifications for: events
  if (key === 'events') {
    const {when, where} = data.event
    const {desc, city, state, country} = where
    let location = city

    if (state) {
      location = `${location}, ${state}`
    }
    if (country) {
      location = `${location} ${country}`
    }

    brief = (
      <div className={styles.event}>
        <Table direction="horizontal">
          <TableBody>
            <TableRow>
              <TableCell>
                <Strong>When</Strong>
              </TableCell>
              <TableCell>
                {getEventTimeDisplay(when)}
              </TableCell>
            </TableRow>
            <TableRow>
              <TableCell>
                <Strong>Where</Strong>
              </TableCell>
              <TableCell>
                <div>{desc}</div>
                <div>
                  {location}
                </div>
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </div>
    )
  }

  return (
    <div className={styles.item}>
      <div className={styles.date}>
        <Time moment={datetime}>{occur}</Time>
      </div>
      <TextLink to={target}>
        <span className={styles.title}>
          {data.title}
        </span>
      </TextLink>
      {brief}
      <div className={styles.author}>
        <Subtle>
          <Avatar name={data.author} />
          {data.author}
          <Spacer />
          {data.org}
        </Subtle>
      </div>
    </div>
  )
}

PostListItem.contextTypes = {
  config: React.PropTypes.object,
}

PostListItem.propTypes = {
  post: React.PropTypes.object.isRequired,
}

export default PostListItem
