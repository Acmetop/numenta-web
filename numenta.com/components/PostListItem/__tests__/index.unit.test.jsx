import React from 'react'
import renderer from 'react-test-renderer'
import stubContext from 'react-stub-context'

import context from '../../../__mocks__/reactContextMock'

import PostListItem from '../../PostListItem'

const _post = {
  file: {
    dir: '/path/to/file/',
  },
  data: {},
  path: '',
}


describe('<PostListItem /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const PostListItemStubbed = stubContext(PostListItem, context)
    const component = renderer.create(
      <PostListItemStubbed post={_post} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
