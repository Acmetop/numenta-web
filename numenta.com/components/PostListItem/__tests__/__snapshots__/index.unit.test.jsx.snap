exports[`<PostListItem /> React component unit test suite Renders correctly 1`] = `
<div
  className="item">
  <div
    className="date">
    <time
      className="time"
      dateTime="Invalid date">
      Invalid date
    </time>
  </div>
  <a
    className="textlink"
    href={undefined}
    onClick={[Function bound handleClick]}
    style={Object {}}
    target={undefined}>
    <span
      className="title" />
  </a>
  <p
    className="paragraph">
    ...
  </p>
  <div
    className="author">
    <span
      className="subtle">
      <span
        className="avatar">
        <img
          alt={undefined}
          className="image border round"
          onClick={undefined}
          src="filename.ext"
          title={undefined} />
      </span>
      <span
        className="spacer">
        â€¢
      </span>
    </span>
  </div>
</div>
`;
